src/controllers/v1/_common/demand.ts

```diff
@@ -94,6 +119,10 @@ export class DemandController extends Controller {
     const [demand] = await this.db.get('demand', { id: demandId, subtype: this.dbDemandSubtype })
     if (!demand) throw new NotFound(this.demandType)
 
+    if (await this.canReadDemand(demand, req)) {
+      throw new Unauthorized('You are not allowed to view this demand')
```

Comment:
A small thought. Given this is a configuration on the service I wonder if this should be a `404` as there's no way of knowing the demand exists even. Probably applies to everything else here